### ISSUE TITLE
There may be a memory leak after grpcio upgrade

### ISSUE THREAD

**Issue by RayWHL** [grpc/grpc#36986]
### What version of gRPC and what language are you using?
grpcio 1.56.2
grpcio 1.60.1
Python 3.10.11

### What operating system (Linux, Windows,...) and version?
CentOS Linux release 7.9.2009 (Core)
kernel 4.18.0-193

### What runtime / compiler are you using (e.g. python version or version of gcc)
Python 3.10.11

### What did you do?
I run [hello-world demo](https://github.com/grpc/grpc/blob/master/examples/python/helloworld/greeter_server.py) in grpcio 1.56.2 and 1.60.1 versions respectively, and monitored the client-side memory usage. Possible memory leak under 1.60.1

server code is: https://github.com/grpc/grpc/blob/master/examples/python/helloworld/greeter_server.py
The client code is based on this file(https://github.com/grpc/grpc/blob/master/examples/python/helloworld/greeter_client.py), and I made some modifications:
```
from __future__ import print_function

import logging

import grpc
import helloworld_pb2
import helloworld_pb2_grpc
import psutil
from datetime import datetime

process = psutil.Process()

def print_memory():
    global process
    memory_info = process.memory_info()
    current_time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")

    print(f"Current Time: {current_time}")
    print(f"RSS (Resident Set Size): {memory_info.rss / (1024 * 1024):.2f} MB")

def run():
    # NOTE(gRPC Python Team): .close() is possible on a channel and should be
    # used in circumstances in which the with statement does not fit the needs
    # of the code.
    print("Will try to greet world ...")
    with grpc.insecure_channel("localhost:50051") as channel:
        stub = helloworld_pb2_grpc.GreeterStub(channel)
        response = stub.SayHello(helloworld_pb2.HelloRequest(name="you"))
    print("Greeter client received: " + response.message)


if __name__ == "__main__":
    logging.basicConfig()
    while True:
        run()
        print_memory()
        # time.sleep(1)
```

### What did you expect to see?
The memory usage in the two versions is approximately the same.

### What did you see instead?
The memory is relatively stable under version 1.56.2, but the memory keeps growing under version 1.60.1

1.56.2
![image](https://github.com/grpc/grpc/assets/31248115/b13b22db-c9de-4ef8-b202-468ebb0ba5dc)

1.60.1
![image](https://github.com/grpc/grpc/assets/31248115/00b3e9de-890d-40ab-b685-1ee14c9dc362)

**Comment by yechuan51** [comment]
I'm seeing the same issue. In my experiment, it appears that just increating the aio insecure channel causes the memory leak. Can you check if this is also the case in your set up?

**Comment by swahnfrayeLC** [comment]
We are also facing the same issue with any version of grpcio > 1.56.2.

**Comment by 9dogs** [comment]
I experimented a bit, and it seems 1.59.5 is the newest version that doesn't leak (the brighter the color the higher the version; the test code is the same @RayWHL used) .
<img width="1201" alt="image" src="https://github.com/user-attachments/assets/b58c4b62-8441-4e27-bdf0-60600bc98f99">
<img width="798" alt="image" src="https://github.com/user-attachments/assets/fb55553b-8cf9-4208-980d-5ddd6594a1d5">

**Comment by sourabhsinghs** [comment]
@RayWHL  Channels should be reused if possible. can you try experimenting with reusing channel across ` run()` .

**Comment by RayWHL** [comment]
Yes, reusing channels does not lead to memory issues, but scenarios involving frequent channel creation can still occur.

**Comment by sourabhsinghs** [comment]
@RayWHL  Memory will increase when creating channel frequently. Whenever garbage collector runs it will cleanup and reclaim memory properly. 
In given script GC will happen only when client actually stops as we are in an infinite loop inside main.

**Comment by 9dogs** [comment]
> Memory will increase when creating channel frequently. Whenever garbage collector runs it will cleanup and reclaim memory properly.
Is there any custom garbage collection logic inside the grpc code? Because CPython itself deallocates memory as soon as the ref count reaches zero. If there are cycle references, then gc will run periodically, but you cannot block it from the Python side (not with a loop).

**Comment by ei-grad** [comment]
Looks like it does not reproduce on latest versions (checked 1.59.x-1.76.0), 1.67.x seems to be last affected with this issue.

<img width="892" height="750" alt="Image" src="https://github.com/user-attachments/assets/fbafa107-8c7e-42e9-9fda-e4c62efd9376" />

**Comment by asheshvidyut** [comment]
Thanks for verifying this @ei-grad 

As the issue is resolved, I am closing this.
